target/linux/ar71xx/files/arch/mips/ath79/mach-e1700ac.c

```diff
@@ -0,0 +1,148 @@
+
+/*
+ * Whqx E1700AC reference board support
+ *
+ * Copyright (c) 2015 Qualcomm Atheros
+ * Copyright (c) 2012 Gabor Juhos <juhosg@openwrt.org>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <linux/platform_device.h>
+#include <linux/ath9k_platform.h>
+#include <linux/ar8216_platform.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-m25p80.h"
+#include "machtypes.h"
+#include "pci.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-spi.h"
+#include "dev-usb.h"
+#include "dev-wmac.h"
+
+#define E1700AC_GPIO_LED_WLAN             12
+#define E1700AC_GPIO_LED_WPS              13
+#define E1700AC_GPIO_LED_STATUS           13
+
+#define E1700AC_GPIO_LED_2G               19
+#define E1700AC_GPIO_LED_DS11             1
+#define E1700AC_GPIO_LED_USB0             7
+
+#define E1700AC_GPIO_BTN_RESET            11
+#define E1700AC_GPIO_BTN_RESET2            2
+
+
+#define E1700AC_KEYS_POLL_INTERVAL        20     /* msecs */
+#define E1700AC_KEYS_DEBOUNCE_INTERVAL    (3 * E1700AC_KEYS_POLL_INTERVAL)
+
+#define E1700AC_MAC0_OFFSET               0
+#define E1700AC_WMAC_CALDATA_OFFSET       0x1000
+
+static struct gpio_led e1700ac_leds_gpio[] __initdata = {
+        {
+                .name           = "e1700ac:green:usb0",
+                .gpio           = E1700AC_GPIO_LED_USB0,
+                .active_low     = 1,
+        },{
+                .name           = "e1700ac:green:ds11",
+                .gpio           = E1700AC_GPIO_LED_DS11,
+                .active_low     = 1,
+        },{
+                .name           = "e1700ac:green:2g",
+                .gpio           = E1700AC_GPIO_LED_2G,
+                .active_low     = 1,
+        },
+};
```

Comment:
I see that this board has five leds, other leds is gpio controlled?

Reply:
@heinzek  this board has five leds，three leds is gpio controlled，two leds are not gpio-controlled
（e.g. power）